src/mpid/ch4/src/ch4_types.h

```diff
@@ -277,6 +277,13 @@ typedef struct {
     unsigned mt_model;
 } MPIDI_CH4_configurations_t;
 
+#define MPIDIU_THREAD_PROGRESS_MUTEX MPIDI_global.m[0]
+#define MPIDIU_THREAD_PROGRESS_HOOK_MUTEX MPIDI_global.m[1]
+#define MPIDIU_THREAD_UTIL_MUTEX MPIDI_global.m[2]
+/* Protects MPIDIG global structures (e.g. global unexpected message queue) */
+#define MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX MPIDI_global.m[3]
+#define MAX_CH4_MUTEXES 4
+
```

Comment:
I guess using numerical directly to access `MPIDI_global.m[]` array is fine as they are assigned only here?

Reply:
Yes. Treating them as `enum` or `#define`s. However, I am not sure tucking them inside the global struct is better than directly use them as globals. Anyway, that is for another time.